### PR TITLE
Curator ignores more than one project with same time and time unit

### PR DESCRIPTION
**Pull request by richm**
[test]

### CODE REVIEW

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
@ewolinetz @sosiouxme PTAL

**Comment by ewolinetz** [comment]
LGTM, do you want to wait until after the default UTC TZ changes are merged in?

**Comment by richm** [comment]
@ewolinetz yes, this commit depends on https://github.com/openshift/origin-aggregated-logging/pull/197

File: curator/run_cron.py [hunk 1]
```diff
@@ -81,7 +86,7 @@
                 value = int(decoded[project][operation][unit])
 
                 if unit in allowed_units:
-                    default_command = default_command + " --exclude " + project + '.*'
+                    default_command = default_command + " --exclude " + shellquote(re.escape(project + '.')) + '*'
```
Comment (sosiouxme) [inline curator/run_cron.py]: shouldn't the '*' be inside the shellquote, but outside the re.escape?

**Comment by sosiouxme** [comment]
So... since project names can't properly include anything but alphanum and hyphen anyway, wouldn't it be better to complain and omit any that do? If someone thinks they can use "foo-*" to represent a bunch of projects, this'll just escape that and match nothing, which isn't what they wanted; while if we give them an error in the logs, they have a better chance of figuring out they did something wrong.

**Comment by richm** [comment]
@sosiouxme Is there a canonical regex that must match project names e.g. `^[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]$`
that is - name must begin and end with alphanum, and may contain hyphens?  If so, I can add a regex to curator to reject names in the config not matching.

**Comment by sosiouxme** [comment]
If you want to be quite correct it needs to be a domain label, lowercase,
no unicode.

```
$ oadm new-project F00
The Project "F00" is invalid.
metadata.name: Invalid value: "F00": must be a DNS label (at most 63
characters, matching regex [a-z0-9]([-a-z0-9]*[a-z0-9])?): e.g. "my-name"
```

However is that the goal here? If we're just trying to filter out project
names that couldn't be correct and could cause parsing problems, surely
`^[-a-z0-9]+$` would do the trick. With `.operations` being an exception
obviously.

On Tue, Jul 12, 2016 at 3:31 PM, Richard Megginson <notifications@github.com

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
@ewolinetz @sosiouxme PTAL

**Comment by ewolinetz** [comment]
LGTM

**Comment by richm** [comment]
depends on https://github.com/openshift/origin-aggregated-logging/pull/184

File: curator/run_cron.py [hunk 2]
```diff
@@ -107,9 +126,15 @@
...
+            tab_command = tab_command + ' --regex '
+            regex = '('
+            sep = ''
             for project in curator_settings[operation][unit][value]:
-                tab_command = tab_command + ' --prefix ' + project + '.'
-
+                regex = regex + sep + '^' + re.escape(project + '.')
+                sep = '|'
+            regex = regex + ')'
+            tab_command = tab_command + shellquote(regex)
```
Comment (sosiouxme) [inline curator/run_cron.py]: this works ok but it feels like it ought to be reduced to a one-liner, something like (not a pythonista):

    tab_command = tab_command + ' --regex ' + shellquote('(' + '|'.join(map(lambda project:'^'+re.escape(project+'.'),curator_settings[operation][unit][value] )) + ')')

OK, maybe two lines, or to taste. I leave it up to you though.

**Comment by sosiouxme** [comment]
nit, otherwise LGTM

**Comment by richm** [comment]
[test]
```error: #1 is already in progress (Running).``` again

**Comment by richm** [comment]
[merge]

**Comment by richm** [comment]
[merge]

**Pull request merged**